src/spring/azext_spring/buildpack_binding.py

```diff
@@ -64,13 +65,27 @@ def buildpack_binding_delete(cmd, client, resource_group, service, name, builder
 
 def create_default_buildpack_binding_for_application_insights(cmd, client, resource_group, name, location,
                                                               app_insights_key, app_insights, sampling_rate):
+    try:
```

Comment:
I don't get why should catch the exception at the very outter place and decide what to do for an internal exception. Can you make the code more comprehensive?

Reply:
At inner code, `_create_app_insights_and_get_connection_string` its purpose is to purely create application insights, it should throw exception when it failed to do it. And it depends on the upper business to decide how to handle such exception.

So here, `create_default_buildpack_binding_for_application_insights`, when it sees such exception, it happens when customer didn't specify anything for application insights, and we create one for cx. The business requirement guides us that we should skip this. So I suppose this method is the place to know how to handle this exception.